### PR TITLE
Create player connected to backend

### PR DESCRIPTION
**Pull request by sebcontreras**
Added hooks and API call to create a new player in the db from the 'join game screen'

### CODE REVIEW

File: src/state/gameSlice.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ const gameStateSlice = createSlice({
   name: "codeSlice",
   initialState: defaultGameState,
   reducers: {
-    setCode: (state: GameState, action: PayloadAction<string>): void => {
+    setGameCode: (state: GameState, action: PayloadAction<string>): void => {
```
Comment (Jhappy77) [inline src/state/gameSlice.ts:14]: I like this change, but we should perhaps change every instance of `code` with `gameCode` to match this. (Except the name `codeSlice` can be `gameSlice` )

File: src/hooks/useCreatePlayer.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+                {
+                    "name": name,
+                    "joinCode": code
+                })
+            .then(response => {
+                // Not sure what we should do with the response
+                console.log(response.status);
+            })
+            .catch(reason => {
+                // TODO: Handle errors better
```
Comment (Jhappy77) [inline src/hooks/useCreatePlayer.ts:25]: This is probably where we will handle the case of the name being already taken. We should test what response the API gives on that scenario (what status code it is), and then send the alert from here

File: src/hooks/useCreatePlayer.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    useEffect(() => {
+        if (!name || !code) return;
+        axios
+            .post(`${API_URL}/players`,
+                {
+                    "name": name,
+                    "joinCode": code
+                })
+            .then(response => {
+                // Not sure what we should do with the response
```
Comment (Jhappy77) [inline src/hooks/useCreatePlayer.ts:21]: My suggestion would be to add a field to the GameSlice like `hasJoinedGame`, and then set it to true when this post request is successful. We can then use the `hasJoinedGame` field to conditionally render stuff later on

**Review by Jhappy77 — CHANGES_REQUESTED** [review]

File: src/state/gameSlice.ts (hunk 1 above)
Comment (sebcontreras) [reply src/state/gameSlice.ts:14]: Gotcha!

File: src/hooks/useCreatePlayer.ts (hunk 2 above)
Comment (sebcontreras) [reply src/hooks/useCreatePlayer.ts:25]: Makes sense, I added an alert here

File: src/hooks/useCreatePlayer.ts (hunk 3 above)
Comment (sebcontreras) [reply src/hooks/useCreatePlayer.ts:21]: Great idea!

**Review by Jhappy77 — APPROVED** [review]

**Pull request merged**